src/components/core/geoChart/GeoChartRenderer.tsx

```diff
@@ -126,28 +128,34 @@ export default class GeoChartRenderer extends React.Component<IGeoChartRendererP
             isExportMode,
             "s-isExportMode": isExportMode,
         });
+
         return <div className={classNames} ref={this.setChartRef} />;
     }
 
-    private updateMapWithConfig = (prevConfig: IGeoConfig): void => {
-        if (this.shouldResetMap(prevConfig)) {
+    private updateMapWithConfig = (prevConfig: IGeoConfig, prevColorStrategy: IColorStrategy): void => {
+        if (this.shouldResetMap(prevConfig, prevColorStrategy)) {
             // Config for clustering and pushpin size lead to change layer setting
             // Then calling resetMap here is needed
             this.resetMap();
         }
-        this.updatePanAndZoom();
         this.updateViewport();
+        this.updatePanAndZoom();
     };
 
     private resetMap = (): void => {
         this.cleanupMap();
         this.setupMap();
     };
 
-    private shouldResetMap = (prevConfig: IGeoConfig): boolean => {
+    private shouldResetMap = (prevConfig: IGeoConfig, prevColorStrategy: IColorStrategy): boolean => {
         if (isPointsConfigChanged(prevConfig.points, this.props.config.points)) {
             return true;
         }
+
+        if (!isEqual(prevColorStrategy.getColorAssignment(), this.props.colorStrategy.getColorAssignment())) {
+            return true;
+        }
```

Comment:
please clean up this fucntion :) 
`const { colorStrategy , config } = this.props;`

Reply:
fixed